server/speak.py

```diff
@@ -9,4 +9,5 @@ def get_speak (text, lang):
     print("speaklang" + lang)
     tts = gTTS(text=text, lang=lang, slow=False)
     tts.save(FILENAME)
-    playsound(FILENAME)
\ No newline at end of file
+
+mp3_path = os.path.join(os.path.dirname(__file__), "tts.mp3")
```

Comment:
I think it will probably blow up !! :D
But you raised a good point, i will give it some thought during this evening and get back to you in the morning...

Reply:
Maybe a queue ?

Reply:
Good thinking! so we'll block the back-end from overwriting the `tts.mp3` until the file is played and then will processes the next request from the queue ? if so, do you think that solution is efficient ? how long would each user waste waiting on another user's TTS to be generated and played until the end 🤔

How about creating a unique random filename for every new TTS request ?

Reply:
Yeah, my idea would work until a certain volume of requests at the same time, after that the wait would be too much. But i've noticed that sometimes similar websites take some time to return the files, even just a few words, don't know if that's the reason..

Your sugestion would decrease the time a lot, it would only take the time to process the request, it could return several files at the same time to each request, but a function to delete the files would be necessary or the folder would have lots of unnecessary files after a while.

A random variable would be needed, like  **pseudocode alert** : 

random = randomUniqueInt, 

and add it to the mp3 filename, something like this : 

static/dist/tts+{randomNumber}.mp3

And clear the folder every day, for example..

What do you think of that ?

I will start working on it !!

Reply:
Awesome, sounds like a plan!